### PR TITLE
[cuda] Query the max block count from the hardware

### PR DESCRIPTION
**Pull request by bobcao3**
Related issue = #

### CODE REVIEW

**Comment by bobcao3** [comment]
Is there a way to trigger an regression benchmark / CI on a PR?

**Review by k-ye — APPROVED** [review]
LGTM!

File: taichi/backends/cuda/cuda_driver.h [hunk 1]
```diff
@@ -30,6 +30,7 @@ constexpr uint32 CU_STREAM_NON_BLOCKING = 0x1;
 constexpr uint32 CU_MEM_ATTACH_GLOBAL = 0x1;
 constexpr uint32 CU_MEM_ADVISE_SET_PREFERRED_LOCATION = 3;
 constexpr uint32 CU_DEVICE_ATTRIBUTE_MAX_BLOCK_DIM_X = 2;
+constexpr uint32 CU_DEVICE_ATTRIBUTE_MAX_BLOCKS_PER_MULTIPROCESSOR = 106;
```
Comment (qiao-bo) [inline taichi/backends/cuda/cuda_driver.h:33]: why 106 here? do we hardcode these values to a specific card or?

**Review by qiao-bo — COMMENTED** [review]
Thanks!

File: taichi/backends/cuda/cuda_driver.h (hunk 1 above)
Comment (bobcao3) [reply taichi/backends/cuda/cuda_driver.h:33]: This is a enum from cuda headers

**Pull request merged**